otter/http.py

```diff
@@ -257,6 +257,11 @@ def perform_tenant_scope(
     to be partially applied, and the result is a performer that can be put into
     a dispatcher.
     """
+    if _concretize is None:
+        # The reason this isn't set as the default kwarg value above is because
+        # of a mock in test_metrics.py.
```

Comment:
So, yeah, the tests that are doing it are the `GnarlyGetMetricsTests`. They are already almost entirely redundant with the tests for `get_all_metrics_effects`, except that they ensure that the thin layer of non-Effect-based code wrapped around it (in `get_all_metrics`) works. That layer is shrinking with this branch, fortunately, but the main reason that function needs to be non-effectful now is because of the whole _perform_limited thing, to maintain limited parallelism while performing the child effects.

There are two solutions to this, one quick, and one better:

- implement our own LimitedParallel intent, which just does what _perform_limited does now in its performer
- add the parallelism limit feature to Effect's `parallel`.

Once we do one of those things, get_all_metrics will be pure and Effectful, and we can get rid of the Gnarly tests. (it also means we'll be able to start refactoring `collect_metrics` to be Effectful, too).